pennylane/_qubit_device.py

```diff
@@ -688,3 +689,107 @@ def sample(self, observable):
         unraveled_indices = [2] * len(device_wires)
         indices = np.ravel_multi_index(samples.T, unraveled_indices)
         return observable.eigvals[indices]
+
+    def rewind_jacobian(self, tape):
+        """Implements the method outlined in https://arxiv.org/abs/2009.02823 to calculate the
+        Jacobian."""
+
+        # Perform the forward pass
+        self.reset()
+        self.execute(tape)
+
+        phi = self._reshape(self.state, [2] * self.num_wires)
+
+        for m in tape.measurements:
+            if m.obs is None:
+                raise ValueError(f"Rewind differentiation method does not support measurement {m}")
+
+            if not hasattr(m.obs, "base_name"):
+                m.obs.base_name = None  # This is needed for when the observable is a tensor product
+
+        lambdas = [self._apply_operation(phi, obs) for obs in tape.observables]
```

Comment:
Regarding supporting probabilities - I think this should be doable, each time we have a `probs` in the return, we'd need to make the series of `2 ** N` projections which would each correspond to a new entry in `lambdas`.

Although for now I'd say let's just do expvals and add appropriate skips in the currently failing tests.

Reply:
:+1:

There is a subtlety here though, for when we make `rewind` default. Currently, it is impossible to choose the default QNode diff method based on tape operations, since we don't know what the tape operations are until QNode execution - and we _need_ to know the diff method before then!

So it would be great to support probabilities when possible.